backend/dataall/modules/s3_datasets_shares/services/s3_share_validator.py

```diff
@@ -122,30 +122,3 @@ def _validate_iam_role_and_policy(
                 action=principal_type,
                 message=f'The principal role {principal_role_name} is not found.',
             )
-
```

Comment:
Removed code is used while creating share object in the graphql lambda calls . I think this is not needed since even if the share policies don't exists and are not created / attached, a share object should be created. 
Once the share object is approved, revoked , then the code for processing the share does always take care of this thing - migrating share policy from inline/ managed ( w/o index ) to managed ( w/ index ) and attaching them at the end.

Reply:
That is not entirely True, if in the share object creation request the customer does not select "attachMissingPolicies" then the managed policies won't be attached during the share task

Reply:
Yeah. You make a good point here. Let me undo those changes